### PR TITLE
Add Mystery Gift mode to SV FCE

### PR DESCRIPTION
**Pull request by NympheaR**

### CODE REVIEW

File: SerialPrograms/Source/PokemonSV/Programs/FastCodeEntry/PokemonSV_CodeEntry.cpp [hunk 1]
```diff
@@ -72,19 +72,22 @@ const char* normalize_code(std::string& normalized_code, const std::string& code
 
     switch (normalized_code.size()){
```
Comment (pifopi) [inline SerialPrograms/Source/PokemonSV/Programs/FastCodeEntry/PokemonSV_CodeEntry.cpp:73]: couldn't you return here if override_mode ?

File: SerialPrograms/Source/PokemonSV/Programs/FastCodeEntry/PokemonSV_FastCodeEntry.cpp [hunk 2]
```diff
@@ -125,6 +126,14 @@ void FastCodeEntry::program(MultiSwitchProgramEnvironment& env, CancellableScope
         return;
     }
 
+    if (MODE == Mode::MYSTERY_GIFT){
+        const char* error = enter_code(env, scope, settings, CODE, true, true);
+        if (error){
```
Comment (pifopi) [inline SerialPrograms/Source/PokemonSV/Programs/FastCodeEntry/PokemonSV_FastCodeEntry.cpp:131]: I don't think this if can ever happen as you always return an empty string

File: SerialPrograms/Source/PokemonSV/Programs/FastCodeEntry/PokemonSV_CodeEntry.cpp (hunk 1 above)
Comment (NympheaR) [reply SerialPrograms/Source/PokemonSV/Programs/FastCodeEntry/PokemonSV_CodeEntry.cpp:77]: added return if override before the other checks

File: SerialPrograms/Source/PokemonSV/Programs/FastCodeEntry/PokemonSV_FastCodeEntry.cpp (hunk 2 above)
Comment (NympheaR) [reply SerialPrograms/Source/PokemonSV/Programs/FastCodeEntry/PokemonSV_FastCodeEntry.cpp]: merged the logic with the normal mode check and have error report only if on normal mode

**Review by pifopi — APPROVED** [review]

**Pull request merged**